### PR TITLE
Handle edit section

### LINKED ISSUE #55: Handle Edit Section
Title: ```PUT/section/:id```
labels: ```back-end``` - ```route```

Description: Edit Section Info

Protected: Yes
Required middleware: ```checkAuth```

Request body:
```JSON
{
    "name": "Section Name Here",
    "projected": 1
}
```

Response
Success: 
```JSON
{
    "data": {
        "id": 1,
        "name": "Section Name Here",
        "projectid": 1
    },
    "message": "Section Updated successfully!"
}
```
& status: 200

Failure: 
```JSON
{
    "message": "SERVER ERROR"
}
```
& status: 500

### PR DESCRIPTION
**Pull request by karamzomlot**
Relates #55
Close #55

### CODE REVIEW

File: server/src/routes/sectionsRoute.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import { addSectionController, deleteSectionController, editSectionController } from '../controllers';
+import { checkAuth } from '../middlewares';
+
+const sectionsRoute = require('express').Router();
+
+sectionsRoute.post('/section', addSectionController);
+sectionsRoute.delete('/section/:id', deleteSectionController);
+sectionsRoute.put('/section/:id', editSectionController);
```
Comment (AmranElmasri) [inline server/src/routes/sectionsRoute.ts:8]: Add checkAuth middleware to your routes

**Review by AmranElmasri — CHANGES_REQUESTED** [review]

**Review by nada-abuzaid — APPROVED** [review]

File: server/src/routes/sectionsRoute.ts (hunk 1 above)
Comment (karamzomlot) [reply server/src/routes/sectionsRoute.ts:8]: If you see the second line of the code, you will find it, but I didn't call it in the route to try the function in postman,
& We will add it after we start with client side

Comment (AmranElmasri) [reply server/src/routes/sectionsRoute.ts:8]: ok

**Review by AmranElmasri — APPROVED** [review]

**Review by MaramMs — APPROVED** [review]

File: server/src/controllers/sections/editSectionController.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+const editSectionController = async (req: Request, res: Response, next: NextFunction) => {
+  const { name, projectId } = req.body;
+  const { id } = req.params;
+  try {
+    // validate user input
+    await addSectionSchema.validateAsync(req.body);
+
+    // update section in database
+    const { rows } = await editSectionQuery(+id, name, +projectId);
```
Comment (Nizar7zak) [inline server/src/controllers/sections/editSectionController.ts]: it is better if we pass all arguments in object here.

**Review by Nizar7zak — CHANGES_REQUESTED** [review]

**Pull request merged**